internal/service/advancedcluster/common_advanced_cluster.go

```diff
@@ -611,3 +613,18 @@ func flattenEndpoints(listEndpoints []matlas.Endpoint) []map[string]any {
 	}
 	return endpoints
 }
+
+func StringIsUppercase() schema.SchemaValidateDiagFunc {
```

Comment:
thanks Andrea!
Question: do we really need one more e2e test for this? Why don't we simply unit test this?

My rationale is: the only business logic that we are responsible for is `StringIsUppercase`; because of that, unit test should suffice. The added value that an e2e can bring here is that we are _effectively confirming_ `ValidateDiagFunc` method is correctly called.. but that should not be our responsibility and we should trust our framework that will do so.

Happy to discuss more about this (I am thinking this from a "invert the test pyramid" perspective)

Reply:
Good point. I will update the PR. Thanks!

Reply:
Updated